tests/describe_task_queue_test.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/stretchr/testify/suite"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/testing/historyrequire"
+	"go.temporal.io/server/common/testing/protorequire"
+	"go.temporal.io/server/common/testing/updateutils"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+// making a new test suite here
+
+type (
+	DescribeTaskQueueSuite struct {
+		*require.Assertions
+		protorequire.ProtoAssertions
+		historyrequire.HistoryRequire
+		updateutils.UpdateUtils
+		FunctionalTestBase
+	}
+)
+
+func (s *DescribeTaskQueueSuite) SetupSuite() {
+	s.dynamicConfigOverrides = map[dynamicconfig.Key]any{
+		dynamicconfig.FrontendEnableWorkerVersioningDataAPIs:     true,
+		dynamicconfig.FrontendEnableWorkerVersioningWorkflowAPIs: true,
+		dynamicconfig.FrontendEnableWorkerVersioningRuleAPIs:     true,
+		dynamicconfig.MatchingForwarderMaxChildrenPerNode:        partitionTreeDegree,
+		dynamicconfig.TaskQueuesPerBuildIdLimit:                  3,
+
+		dynamicconfig.AssignmentRuleLimitPerQueue:              10,
+		dynamicconfig.RedirectRuleLimitPerQueue:                10,
+		dynamicconfig.RedirectRuleChainLimitPerQueue:           10,
+		dynamicconfig.MatchingDeletedRuleRetentionTime:         24 * time.Hour,
+		dynamicconfig.ReachabilityBuildIdVisibilityGracePeriod: 3 * time.Minute,
+		dynamicconfig.ReachabilityQueryBuildIdLimit:            4,
+
+		// Make sure we don't hit the rate limiter in tests
+		dynamicconfig.FrontendGlobalNamespaceNamespaceReplicationInducingAPIsRPS:                1000,
+		dynamicconfig.FrontendMaxNamespaceNamespaceReplicationInducingAPIsBurstRatioPerInstance: 1,
+		dynamicconfig.FrontendNamespaceReplicationInducingAPIsRPS:                               1000,
+
+		// The dispatch tests below rely on being able to see the effects of changing
+		// versioning data relatively quickly. In general, we only promise to act on new
+		// versioning data "soon", i.e. after a long poll interval. We can reduce the long poll
+		// interval so that we don't have to wait so long.
+		// TODO: update this comment. it may be outdated and/or misleading.
+		dynamicconfig.MatchingLongPollExpirationInterval: longPollTime,
+
+		// this is overridden for tests using testWithMatchingBehavior
+		dynamicconfig.MatchingNumTaskqueueReadPartitions:  4,
+		dynamicconfig.MatchingNumTaskqueueWritePartitions: 4,
+	}
+	s.setupSuite("testdata/es_cluster.yaml")
+}
+
+func (s *DescribeTaskQueueSuite) TearDownSuite() {
+	s.tearDownSuite()
+}
+
+func (s *DescribeTaskQueueSuite) SetupTest() {
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+	s.ProtoAssertions = protorequire.New(s.T())
+	s.HistoryRequire = historyrequire.New(s.T())
+	s.UpdateUtils = updateutils.New(s.T())
```

Comment:
Not sure if we need these in this Suite. I see you are validating history events somewhere but I don't think you need to as that is not a goal of these tests. There are other tests suites for that.